File: IBAJavaProject/src/threads/RecordsThread.java
package threads;

import models.Appointment;
import models.MedicalCenter;

import java.util.List;
import java.util.Random;

public class RecordsThread extends Thread{

    private MedicalCenter medicalCenter;
    public RecordsThread(String name, MedicalCenter medicalCenter) {
        super(name);
        this.medicalCenter = medicalCenter;
    }

    public void run() {
        System.out.println(Thread.currentThread().getName() + " started...\n");

        long beginTime = System.nanoTime();
        final long MINUTE = 60 *1000 * 1000 * 1000L;
        long endTime = beginTime + MINUTE;
        while(System.nanoTime() <= endTime) {
            int time = (int)(( Math.random() * (10 - 0 + 1) + 0));

            try {
                Thread.sleep(time*1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }

            Random random = new Random();
            Appointment record = new Appointment(
                    "Patient" + (medicalCenter.getCountOfRecords() + 1),
                    random.nextBoolean(),
                    this.medicalCenter.getLastArrivalTime() + time
            );
            System.out.println(
                    record.getFirstName() + " " +
                            record.ticketAvailability() + " arrived to " +
                            record.getTimeArrived() + "\n"
            );
            medicalCenter.therapistTraffic(record);

            System.out.println("Doctor " + medicalCenter.getDoctorsName() + " is there");
            List<Appointment> queue = medicalCenter.getQueue();
            for (Appointment _record : queue) {
                System.out.println(_record);
            }
            System.out.println();
        }
        System.out.println(Thread.currentThread().getName() + " finished...");
    }
}

File: IBAKotlinProject/src/main/kotlin/threads/RecordsThread.kt
package threads

import models.Appointment
import models.MedicalCenter
import java.util.*

class RecordsThread(name: String?, private val medicalCenter: MedicalCenter) : Thread(name) {
    override fun run() {
        println(currentThread().name + " started...\n")
        val beginTime = System.nanoTime()
        val MINUTE = 60 * 1000 * 1000 * 1000L
        val endTime = beginTime + MINUTE
        while (System.nanoTime() <= endTime) {
            val time = (Math.random() * (10 - 0 + 1) + 0).toInt()
            try {
                sleep((time * 1000).toLong())
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
            val random = Random()
            val record = Appointment(
                "Patient" + (medicalCenter.countOfRecords + 1),
                random.nextBoolean(),
                medicalCenter.lastArrivalTime + time
            )
            println(
                record.firstName + " " +
                        record.ticketAvailability() + " arrived to " +
                        record.timeArrived + "\n"
            )
            medicalCenter.therapistTraffic(record)
            println("Doctor " + medicalCenter.doctorsName + " is there")
            val queue = medicalCenter.getQueue()
            for (_record: Appointment? in queue) {
                println(_record)
            }
            println()
        }
        println(currentThread().name + " finished...")
    }
}